Enh(mp)monitoring proc before release (#284)
centreon/centreon-documentation: en/integrations/plugin-packs/procedures/network-dell-xseries-snmp.md

| cpu.utilization.1s.percentage            | CPU usage on 1s.       |   %   |
| cpu.utilization.1m.percentage            | CPU usage on 1m.       |   %   |
| cpu.utilization.5m.percentage            | CPU usage on 5m.       |   %   |

<!--Interfaces-->

| Metric name                              | Description                                                | Unit   |
| :--------------------------------------- | :----------------------------------------------------------| :----- |
| status                                   | Status of the interface                                    | String |
| interface.traffic.in.bitspersecond       | Incoming traffic going through the interface.     			| Bits/s |
| interface.traffic.out.bitspersecond      | Outgoing traffic going through the interface.    			| Bits/s |
| interface.packets.in.error.percentage    | Incoming errored packets going through the interface. 		|   %	 | 
| interface.packets.in.discard.percentage  | Incoming discarded packets going through the interface.	|   %	 |
| interface.packets.out.error.percentage   | Outgoing errored packets going through the interface. 		|   %	 |

> | interface.packets.out.discard.percentage | Outgoing discarded packets going through the interface. 	|   %	 | 

Reviewer: Pandoc format

<!--Uptime-->

| Metric name | Description                                | Unit |
| :---------- | :----------------------------------------- | :--: |
| uptime      | Elapsed time since last reboot             |   s  |

<!--END_DOCUSAURUS_CODE_TABS-->

### Prerequisites

### Dell Xseries Device configuration

The SNMP v2 agent has to be configured on the Dell Xseries device.

### Network flow

> The Centreon Poller should be able to reach Dell Xseries device over UDP/161 SNMP port

Reviewer: ```suggestion
The Centreon Poller should be able to reach the UDP/161 SNMP port of the Dell Xseries device.

```

```

2. Install the Centreon Plugin-Pack RPM on the Centreon Central server:

```bash
yum install centreon-pack-network-switch-dell-xseries-snmp
```

3. Install the *Dell Xseries* Centreon Plugin-Pack from the "Configuration > Plugin packs > Manager" page

<!--END_DOCUSAURUS_CODE_TABS-->

## Configuration

* In the Centreon Web interface, add a new Host from the "Configuration > Hosts" page.
* Set the values of the *SNMP community* and *SNMP version* fields then apply the *Net-Switch-Dell-Xseries-SNMP-custom* Host template.

> If you're using the version 3 of the SNMP protocol, select the related SNMP version in the Host configuration form and
> set the SNMP v3 specific settings in the *SNMPEXTRAOPTIONS* Macro:

| Mandatory   | Name                    | Description                       |
| :---------- | :---------------------- | :---------------------------------|
|             | SNMPEXTRAOPTIONS        | Extra options SNMP of Dell Xseries|

## FAQ

### How can I test the Plugin in the CLI and what do the main options mean?

Once the Plugin is installed, you can test it directly from the command line interface of the Centreon Poller with the *centreon-engine* user:

```bash

> /usr/lib/centreon/plugins/centreon_dell_xseries_snmp.pl
> 	--plugin=network::dell::xseries::snmp::plugin
> 	--hostname=localhost
> 	--snmp-version='2c'
> 	--snmp-community='public' 
> 	--mode=cpu
> 	-warning-average-1s='80' 
> 	--critical-average-1s='90'
> 	--warning-average-1m='80' 
> 	--critical-average-1m='90'
> 	--verbose

Reviewer: ```suggestion
/usr/lib/centreon/plugins/centreon_dell_xseries_snmp.pl \
	--plugin=network::dell::xseries::snmp::plugin \
	--hostname=10.0.0.1 \
	--snmp-version='2c' \
	--snmp-community='public' \
	--mode=cpu \
	--warning-average-1s='80' \
	--critical-average-1s='90' \
	--warning-average-1m='80' \ 
	--critical-average-1m='90' \
	--verbose

```

```

The expected result should output something similar to:

```bash
OK: cpu total: 15 % average-1s: 18.00% average-1m: 25.00% average-5m: 15.00%|
'cpu.utilization.1s.percentage'=18%;80;90;0;100; 'cpu.utilization.1m.percentage'=25%;80;90;0;100; 'cpu.utilization.1s.percentage'=15%;;;0;100
```

The above command checks a Dell Xseries switch using the SNMP protocol (```--plugin=network::dell::xseries::snmp::plugin```) 
with the *public* community (```--snmp-community='public'```) and the *2c* SNMP version (```--snmp-version='2c'```).
This command checks the current CPU average of the switch (```--mode='cpu'```).

> This command will trigger a WARNING alarm if the average on 1s increases over 80% (```--warning-average-1s='80'```)

Reviewer: ```suggestion
This command will trigger a WARNING alarm if the average on usage per seconde increases over 80% (```--warning-average-1s='80'```)

```
